src/stores/FlairStore.js

```diff
@@ -149,13 +152,28 @@ class FlairStore {
             return this._groupProfiles[groupId];
         }
 
-        const profile = await matrixClient.getGroupProfile(groupId);
+        // No request yet, start one
+        if (!this._groupProfilesPromise[groupId]) {
+            this._groupProfilesPromise[groupId] = matrixClient.getGroupProfile(groupId);
+        }
+
+        let profile;
+        try {
+            profile = await this._groupProfilesPromise[groupId];
+        } catch (e) {
+            // Don't retry, but allow a retry when the profile is next requested
```

Comment:
do we perhaps want to log the exception here, just in case it's something unexpectedly bad?

Reply:
Yep, seems fair